SSR-1580: Create contributing guide and pull request template (#898)
kintone-labs/kintone-ui-component: CONTRIBUTING.md

The format of the Branch Name is as follows  
```sh
[Version No]_Title
```
For example
```sh
origin/master/v1.1.0
  |__origin/master/v1.1.0_Button_src
  |__origin/master/v1.1.0_Button_unit_test
  |__origin/master/v1.1.0_Button_docs
  |__origin/master/v1.1.0_Notification_docs  
  |__origin/master/v1.1.0_Text_src

origin/v0_dev/v0.8.3
  |__origin/v0_dev/v0.8.3_Button_src
  |__origin/v0_dev/v0.8.3_Button_unit_test
  |__origin/v0_dev/v0.8.3_Button_docs
  |__origin/v0_dev/v0.8.3_Notification_docs  
  |__origin/v0_dev/v0.8.3_Text_src
```
### Pull Request

If you finish the implementation of the child branches, you should merge from the parent branch before creating a pull request against it.  

The format of the PR title is as follows  
```sh
[PR Type] PR Title
```

PR types are as follows  
| Type |	Example |
| :---- | :---- |
| Bugfix |	[BugFix] Fix a bug |
| HotFix |	[HotFix] Urgently fix something |
| CS | [CS] Fix Coding Style |
| Feature | [Feature] Wrote a new feature |
| Update | [Update] Changed a doc file |
| Refactor | [Refactor] Refactored a certain function |

### Issue

> The format of the issue title is as follows  

Author: ```suggestion
The format of the Issue Title is as follows:  
```

```sh

> [Issue Type] Issue Title  

Author: ```suggestion
[Issue Type] Title  
```

```

> |Type| Example|
> |:----|:----|
> |Bug report| [Bug report] Found a bug about XXX |
> |Feature request|[Feature request] Suggest an idea about XXX|

Author: ```suggestion
|Type| Example|
|:----|:----|
|BugReport| [BugReport] Found a bug about XXX |
|FeatureRequest|[FeatureRequest] Suggest an idea about XXX|
```

# Contributing Guide

## For Contributors

Contributions are always welcome!
If you have discovered a bug or have a feature request, [please create an issue on GitHub](https://github.com/kintone-labs/kintone-ui-component/issues/new/choose).  
Pull requests are also welcome when you find trivial bugs or typos.

We use English for all commit messages, code comments, issues, pull requests.

### Setup

> This repository is a monorepo using npm.

Reviewer: This means that there is only one project in this repository.
https://css-tricks.com/from-a-single-repo-to-multi-repos-to-monorepo-to-multi-monorepo/

> This repository is a monorepo using npm.

Author: @h000780 
On that point, I think it won't be monorepo but single repo. How do you think?

```sh
% cd kintone-ui-component
% npm ci
```

> ### Build
> The project should be Setup before Build.
> ```sh
> % cd kintone-ui-component
> % npm run build:umd_prod
> ```
> 
> ### Development
> The project should be Setup before Development.
> 
> ```sh
> % cd kintone-ui-component
> % npm run build:umd_dev
> ```

Author: I think we can summarize like this.

```suggestion
### Build
The project should be Setup before Build.

#### UMD (prod mode)
```sh
% cd kintone-ui-component
% npm run build:umd_prod
```

#### UMD (dev mode)
```sh
% cd kintone-ui-component
% npm run build:umd_dev
```

#### ESM
```sh
% cd kintone-ui-component
% npm run build:esm
```
```

### Unit Test
Unit test confirm DOM is correct and that if necessary throw error.  
```sh
% cd kintone-ui-component
% npm run test
```

> ### Storybook
> Storybook Controls gives you a graphical UI to interact with a component's arguments dynamically, without needing to code. It creates an addon panel next to your component examples ("stories"), so you can edit them live.  

Author: How about writing like this and moving this section to the above of Build section? (order by actual dev step)
```suggestion
### Develop with Storybook
You can develop and check the behavior graphically with Storybook.
Storybook Controls gives you a graphical UI to interact with a component's arguments dynamically, without needing to code. It creates an addon panel next to your component examples ("stories"), so you can edit them live.
```

    { page: 'help', label: 'Help' },
    /* you can add external links */
    { href: 'https://github.com/facebook/docusaurus', label: 'GitHub' },
    ...
  ],
  ...
}
```
##### Adding custom page
docusaurus uses React components to build a page.  
components are saved as JS files in website/pages/en  
###### 1.Create a page in website/pages/en
###### 2.Specify this page in the headerLinks of website/siteConfig.js If you want to link it to the top navigation bar.
```sh
{
  headerLinks: [
    ...
    { page: 'my-new-custom-page', label: 'My New Custom Page' },
    ...
  ],
  ...
}
```
#### Build
##### 1.Move to  website/directory, run npm run build to build
```sh
cd website/
npm run build
```
##### 2. Verify that a build/directory is created and it contains the .html of other pages and all the documents in the website/ directory

#### Versioning
##### 1.After moving to the website/ directory, specify the version’s number to generate the document of this corresponding version.
```sh
cd website/
npm run version <version number>
```
##### 2.Verify that website/versioned_docs/version-{version number} and website/versioned_sidebars/version-{version number}-sidebars.jsonare created.

> ##### 3.Verify the specified version is added in website/versions.json

Author: This section is too long, so I think we can add a link to this. (maybe need to update the target content though)
https://github.com/kintone-labs/kintone-ui-component/tree/master/docs/document

For example
```sh
origin/master/v1.1.0
  |__origin/master/v1.1.0_Button_src
  |__origin/master/v1.1.0_Button_unit_test
  |__origin/master/v1.1.0_Button_docs
  |__origin/master/v1.1.0_Notification_docs  
  |__origin/master/v1.1.0_SSR-xxx
```

#### 2.Pull Request Naming Rule
If you finish the implementation of the child branches, you should merge from the parent branch before creating a pull request against it.  

The format of the PR title is as follows  
```sh
[PR Type] (Task Number) PR Title
```

PR types are as follows  
| Type |	Example |
| :---- | :---- |
| Bugfix |	[BugFix] Fix a bug |
| HotFix |	[HotFix] Urgently fix something |
| CS | [CS] Fix Coding Style |
| Feature | [Feature] Wrote a new feature |
| Update | [Update] Changed a doc file |
| Refactor | [Refactor] Refactored a certain function |

#### 3.Issue Naming Rule
The format of the issue title is as follows  
```sh
[Issue Type] Issue Title  
```
|Type| Example|
|:----|:----|
|Bug report| [Bug report] Found a bug about XXX |
|Feature request|[Feature request] Suggest an idea about XXX|

> 

Author: I think we should make it compact. Let's discuss directly.

### Clone and create branch

#### 1. Clone from GitHub repository [here](https://github.com/kintone-labs/kintone-ui-component) to local

```sh
git clone git@github.com:kintone-labs/kintone-ui-component.git
```

#### 2. Checkout to branch for development
Please refer to [Branch naming rule](#Branch) and create a new branch for development from the next version branch.
```sh
git branch <branch name>
git checkout <branch name>
```
### Setup

> This repository is a Singgle-repo using npm.

Author: Please fix the misspelling.
> Single Repo

```sh
cd kintone-ui-component
npm ci
```
### Develop
#### 1. Implement your functions/features
#### 2. ESlint check
```sh
npm run es-lint
```
#### 3. Start storybook to confirm your functions/features
```sh
npm run storybook
```
#### 4. Commit and Create Pull Request
Please refer to [Pull Request naming rule](#Pull-Request).  

### Build
```sh
npm run build:umd_dev

> or

Author: Why did you add only umd_dev not including esm??
My suggestion was below. Please check it.
https://github.com/kintone-labs/kintone-ui-component/pull/898#discussion_r796604907

npm run build:umd_prod
or
npm run build:esm
```
### Update document
Please refer to [Document with Docusaurus](https://github.com/kintone-labs/kintone-ui-component/blob/master/docs/document/README.md).

## Note for Naming Rule

### Branch

You can create several other child branches to implement source code for functions/features and/or unit tests and/or user guides.

The format of the Branch Name is as follows:  
```sh
{Version Number}_Title
```
For example:
```sh
origin/master/v1.1.0
  |__origin/master/v1.1.0_Button_src
  |__origin/master/v1.1.0_Button_unit_test
  |__origin/master/v1.1.0_Button_docs
  |__origin/master/v1.1.0_Notification_docs  
  |__origin/master/v1.1.0_Text_src
```
### Pull Request

If you finish the implementation under the child branches, you should merge from the parent branch before creating a pull request against it.  

The format of the Pull Request Title is as follows:  
```sh
[Pull Request Type] Title
```

> Pull Request types are as follows  

Author: Please add `:` at the end of this line.

| Type | Example |
| :---- | :---- |
| Bugfix | [BugFix] Fix a bug |
| HotFix | [HotFix] Urgently fix something |
| CS | [CS] Fix coding style |
| Feature | [Feature] Write a new feature |
| Update | [Update] Change a doc file |
| Refactor | [Refactor] Refactor a certain function |

### Issue

> The format of the Issue Title is as follows:s  

Author: Please delete the unnecessary `s` at the end.
